lib/services/sorting/post/post_sorting_service.dart

```diff
@@ -25,31 +25,27 @@ class PostSortingService {
           .compareTo(option.scoreFunction(postB, position));
       // Multiplying by -1 reverses the order
       final sign = option.sortIncreasing ? 1 : -1;
+
       return sign * order;
     }
 
     late final Comparator<PostFirestore> comparator;
-    if (putOnTop.isEmpty) {
-      // This is a slight optimisation that allows the function
-      // to have the expected complexity when putOnTop is empty
-      // (though having putOnTop non empty does not change the asymptotic
-      // complexity of the function if it is a hash set/of constant size)
-      comparator = defaultComparator;
-    } else {
-      comparator = (a, b) {
-        final putOnTopA = putOnTop.contains(a.id);
-        final putOnTopB = putOnTop.contains(b.id);
-
-        if (putOnTopA && !putOnTopB) {
-          // a is considered smaller
-          return -1;
-        } else if (!putOnTopA && putOnTopB) {
-          return 1;
-        }
-
-        return defaultComparator(a, b);
-      };
-    }
+
+    putOnTop.isEmpty
+        ? comparator = defaultComparator
+        : comparator = (a, b) {
```

Comment:
The assignment should be done outside of the ternary operator to avoid repetition: `comparator = ... ? ... : ...`

We might also want a local private function definition instead of inlining it like that (for the else clause).

Reply:
I tried a new implementation with [9886fa4](https://github.com/ProximaEPFL/proxima/pull/262/commits/9886fa47af081891b783416283bf94f11ca43052). What's your opinion on it?

Reply:
Much better, ty :+1: